plugins/content/joomla/src/Extension/Joomla.php

```diff
@@ -210,6 +216,406 @@ public function onContentBeforeChangeState($context, $pks, $value)
+            $this->injectContactSchema($context, $schema);
+        }
+    }
+
+    /**
+     * Inject com_content schemas if needed
+     *
+     * @param   string    $context  The com_content context like com_content.article.5
+     * @param   Registry  $schema   The overall schema object to manipulate
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    private function injectContentSchema(string $context, Registry $schema)
+    {
+        $app = $this->getApplication();
+        $db = $this->getDatabase();
+
+        list($extension, $view, $id) = explode('.', $context);
+
+        // Check if there is alrady a schema for the item, then skip it
+        $mySchema = $schema->toArray();
+
+        if (!isset($mySchema['@graph']) || !is_array($mySchema['@graph'])) {
+            return;
+        }
+
+        $baseId = Uri::root() . '#/schema/';
+        $schemaId = $baseId . str_replace('.', '/', $context);
+
+        foreach ($mySchema['@graph'] as $entry) {
+            // Someone added our context already, no need to add automated data
+            if (isset($entry['@id']) && $entry['@id'] == $schemaId) {
+                return;
+            }
+        }
+
+        $additionalSchema = [];
+
+        $component = $this->getApplication()->bootComponent('com_content')->getMVCFactory();
+
+        $enableCache = $this->params->get('schema_cache', 1);
+
+        $cache = Factory::getContainer()->get(CacheControllerFactory::class)
+            ->createCacheController('Callback', ['lifetime' => $app->get('cachetime'), 'caching' => $enableCache, 'defaultgroup' => 'schemaorg']);
+
+        // Add article data
+        if ($view == 'article' && $id > 0) {
+            $additionalSchema = $cache->get(function ($id) use ($component, $baseId) {
+                $model = $component->createModel('Article', 'Site');
+
+                $article = $model->getItem($id);
+
+                if (empty($article->id)) {
+                    return;
+                }
+
+                $article->images = new Registry($article->images);
+
+                $articleSchema = $this->createArticleSchema($article);
+
+                $articleSchema['isPartOf'] = ['@id' => $baseId . 'WebPage/base'];
+
+                return $articleSchema;
+            }, [$id]);
+        } elseif (in_array($view, ['category', 'featured', 'archive'])) {
+            $additionalSchema = $cache->get(function ($view, $id) use ($component, $baseId, $app) {
+                $menu = $app->getMenu()->getActive();
+                $schemaId = $baseId . 'com_content/' . $view . ($view == 'category' ? '/' . $id : '');
+
+                $additionalSchema = [
+                    '@type'    => 'Blog',
+                    '@id'      => $schemaId,
+                    'isPartOf' => ['@id' => $baseId . 'WebPage/base'],
+                    'name'     => htmlentities($menu->title),
+                    'blogPost' => []
```

Comment:
```suggestion
                    'blogPost' => [],
```
PHPCS comma after last array element